src/worker/activities/activity_heartbeat_manager.rs

```diff
@@ -105,143 +111,221 @@ impl ActivityHeartbeatManager {
 }
 
 #[derive(Debug)]
-struct HbStreamState {
-    last_sent: HashMap<TaskToken, ActivityHbState>,
-    incoming_hbs: UnboundedReceiver<HBAction>,
+struct ActivityHeartbeatState {
+    task_token: TaskToken,
+    last_recorded_details: Option<Vec<common::Payload>>,
```

Comment:
This probably could do without the `Option` since an empty `Vec` isn't worth reporting either.